### DISCUSSION TITLE
23.01 and pfSense-API v1.6.0_devc01a130?

### DISCUSSION THREAD

**Discussion by HovKlan-DH** [jaredhendrickson13/pfsense-api#338] [Q&A] [answered]
I am eager to try out this API, so I can control my firewall rules via automation, but I am unsure, if I will be able to use the 1.6.0 development version for my Netgate 1100 with the 23.01-RELEASE?

**Comment by jaredhendrickson13** [comment, marked as answer]
Yes and no. Netgate hardware uses an ARM build of pfSense that isn't publicly available. I don't have access to the ARM build of pfSense Plus so I don't have a reliable way of building and testing the package on ARM. Because of this it's _technically_ not supported. It is also something that is not currently supported by Netgate, so installing would be at your own risk.

That being said, while I personally wouldn't recommend it, you should be able to force the package to be installed on the ARM build (#239). The package _should_ work the same between both architectures; I haven't heard of any major functional issues between the package on ARM64 vs AMD64 besides what I've mentioned in issue #239. If you do, I'd make sure you have backups and maybe take a ZFS snapshot before installing.

**Comment by HovKlan-DH** [comment]
Now there is a new API v2 - has this changed anything or is it still not advisable to install **API v2** on my **Netgate 1100** with **ARM64** and its newest release, **24.03-RELEASE**? :-)

I am asking because on the v2.0.1 announcement I can see this statement, "_This change should only affect Netgate 3100 users. It is recommended affected users remain on v1._", so I am now in doubt.

It would be really nice, if you could post a statement on the front page of this project, to clarify exactly what Netgate owners _should_ or should _not_ do in terms of this API. I really want an API, but then again I do not want to ruin my Netgate 1100 by installing this.

**Reply by jaredhendrickson13** [reply]
Yes, since [v1.6.3](https://github.com/jaredhendrickson13/pfsense-api/releases/tag/v1.6.3) the package has been built to install cleanly on arm64 systems. This means you no longer need to force the installation of the package if you choose to use it on arm64. This also allows the package's update features work the same as amd64 systems.

However, unofficial packages are still not endorsed or supported by Netgate. Therefore, using this package (mostly in regards to pfSense Plus) is at your own risk in terms of support and acceptable use.

There isn't anything specific that users of arm64 devices should or need to do differently compared to users on amd64. I just can't guarantee compatibility for Netgate/arm64 devices between releases since I do not have an arm64 environment to perform automated testing on (as noted in the [install requirements](https://pfrest.org/INSTALL_AND_CONFIG/#hardware-requirements)).

I do rely heavily on the community to report and assist with issues specific to Netgate devices and pfSense Plus in general. These issues are addressed whenever possible (like issue #502). I am aware of at least a dozen successful installations on Netgate devices, and there are likely many more.

To directly address your concern: If your Netgate 1100 is running the newest release, 24.03-RELEASE, the v2 package should install and function the same as an amd64 system of equivalent spec. I'd still recommend having backups and/or a ZFS snapshot just in case.

**Reply by HovKlan-DH** [reply]
Hmm.. doing this from the [install documentation](https://pfrest.org/INSTALL_AND_CONFIG/):

`[24.03-RELEASE][admin@router.domain]/root: pkg-static -C /dev/null add https://github.com/jaredhendrickson13/pfsense-api/releases/latest/download/pfSense-24.03-pkg-RESTAPI.pkg`

.. which results in this:

```
[24.03-RELEASE][admin@router.domain]/root: pkg-static -C /dev/null add https://github.com/jaredhendrickson13/pfsense-api/releases/latest/download/pfSense-24.03-pkg-RESTAPI.pkg
Fetching pfSense-24.03-pkg-RESTAPI.pkg: 100%  751 KiB 768.5kB/s    00:01    
Installing pfSense-pkg-RESTAPI-2.0_3...
pkg-static: wrong architecture: FreeBSD:15:amd64 instead of FreeBSD:15:aarch64

Failed to install the following 1 package(s): https://github.com/jaredhendrickson13/pfsense-api/releases/latest/download/pfSense-24.03-pkg-RESTAPI.pkg
```

My system is this one here:

Netgate 1100
24.03-RELEASE (arm64)
FreeBSD 15.0-CURRENT
CPU type: ARM Cortex-A53 r0p4

Am I doing anything wrong here?

**Reply by jaredhendrickson13** [reply]
Hmm no you look to be doing everything correctly. `aarch64` is marked as a supported architecture in the package's [Makefile](https://github.com/jaredhendrickson13/pfsense-api/blob/82b6baeb3e7dd30f4808f568992adff2521271ea/tools/templates/Makefile.j2#L17C1-L17C29) so I'm not sure why it'd be complaining. Let me do some testing and get back to you.

**Reply by jaredhendrickson13** [reply]
I found an issue with my FreeBSD 15 build server that likely caused this. I rebuilt v2.0.3 and updated the release. Give it another try and let me know if you still see the issue occur.

**Reply by HovKlan-DH** [reply]
This helped :-)

```
[24.03-RELEASE][admin@router.domain]/root: pkg-static -C /dev/null add https://github.com/jaredhendrickson13/pfsense-api/releases/latest/download/pfSense-24.03-pkg-RESTAPI.pkg
Fetching pfSense-24.03-pkg-RESTAPI.pkg: 100%  752 KiB 770.4kB/s    00:01    
Installing pfSense-pkg-RESTAPI-2.0_3...
Extracting pfSense-pkg-RESTAPI-2.0_3: 100%
Saving updated package information...
done.
Loading package configuration... done.
Configuring package components...
Loading package instructions...
Custom commands...
Menu items... done.
Writing configuration... done.
Building endpoints... done.
Building privileges... done.
Building forms... done.
Building privileges... done.
Generating OpenAPI documentation... done.
Restoring REST API configuration... no backup found.
Starting REST API settings sync... not configured.
Configuring schedule for \RESTAPI\Caches\AvailablePackageCache... done.
Configuring schedule for \RESTAPI\Caches\RESTAPIVersionReleasesCache... done.
Initiating webConfigurator restart... done.
[24.03-RELEASE][admin@router.domain]/root:
```

Thanks for your quick help, and I will now dive into the API.
